azext_iot/tests/digitaltwins/test_dt_twin_unit.py

```diff
@@ -420,6 +433,12 @@ def test_update_twin_error(self, fixture_cmd, service_client_error):
                 resource_group_name=None,
                 etag=None
             )
+        status_codes = [call.response.status_code for call in service_client_error.calls]
+        # Retries 5 times
```

Comment:
What about having the service_client_error fixture patch the service client so it doesn't retry?

Reply:
In a real service call, a 500 ends with the retries so this is closer to a real life situation. Plus the retries in the unit test are still very fast.

Reply:
Why is it not consistent where all other service_client_error based unit tests don't require this same setup for retries? And why does updating responses necessitate this?

Reply:
hmm ok let me look through the other retries and update them so the tests look for the retry calls

Reply:
It's more about understanding the why this behavior in _this responses version_ before applying it everywhere.